packages/client/src/components/OrderPayment/OrderPayment.tsx

```diff
@@ -38,17 +50,27 @@ const OrderPayment = () => {
   const [securityCode, setSecurityCode] = useState('');
   const securityCodeErrorMessage = securityCode?.match(CODE_REGEX) ? '' : 'Wpisz poprawny kod.';
 
+  const [cardDataEdited, setCardDataEdited] = useState(false);
+
   const [didSubmit, setDidSubmit] = useState(false);
   const [userData, setUserData] = useState<UserDto | undefined>();
+  const [modalText, setModalText] = useState('Dziękujemy za skorzystanie z naszego serwisu. Smacznego!');
+  const [modalTitle, setModalTitle] = useState('Zamówienie zostało złożone');
+
+  const [open, setOpen] = useState(false);
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
 
-  const [cardDataEdited, setCardDataEdited] = useState(false);
   const ordersApi = new OrdersApi(apiConfiguration);
+
   const userOrder = async (updateData: CreateOrderDto) => {
     try {
       await ordersApi.create({ createOrderDto: updateData });
     } catch (e) {
       // eslint-disable-next-line no-console
       console.error(e);
+      setModalTitle('Ooops, wystąpił błąd');
+      setModalText('Prosimy spróbować później...');
```

Comment:
Jeżeli ustawimy tu stan na komunikaty o błędzie, to nawet jeśli następne zapytanie się uda (bo odlaguje backend czy coś) będziemy mieli dalej komunikat o niepowodzeniu.

Najłatwiej do `try`, pod zapytanie dodać `set...` ustawiające tą wiadomość o sukcesie, którą dałeś w `useState` (i wtedy można do `useState` podawać tylko `""`, bo komunikat i tak będzie aktualizowany przy każdym zamawianiu).

Reply:
z lekkim opóźnieniem działa set... i teraz jest moment, że popup nie ma tekstu :/

Reply:
Chyba zostawię tak jak jest, a dodatkowo do try dodam znowu ustawienie tego poprawnego tekstu?

Reply:
> z lekkim opóźnieniem działa set...

Nawet nie tyle jest opóźnienie na secie, co na wysyłaniu orderu. Wyświetlamy wynik operacji (złożenia zamówienia) zanim ta operacja się wykona i dopiero po wykonaniu tekst "dogania" otworzenie popupa.

Najprościej chyba otworzyć popup dopiero po tym kiedy znamy wynik wysłania orderu (poprawny/błąd), czyli w `handleSubmit`, dodać `await` do wywołania `userOrder` (i zrobić z funkcji `async` żeby można tak było) albo zrobić `userOrder(...).then(() => handleOpen())`.